sdk/assemblyscript/src/models/openai/chat.ts

```diff
@@ -317,26 +317,20 @@ export class ToolChoice {
     return new ToolChoice("function", name);
   }
 
-  __INITIALIZE(): this {
-    return this;
-  }
 
-  __SERIALIZE(): string {
-    if (this.type == "function") {
+  @serializer
+  serialize(self: ToolChoice): string {
+    if (self.type == "function") {
```

Comment:
I'm confused.  You've replaced `this` with `self` here, but `this` is still being used in the remainder of the function and it still works.  I can also replace other uses of `self` with `this` and everything seems to work.  So why have `self` at all then?

Reply:
Hm, I suppose `self` isn't required for custom serializers, but it is required for the actual `__SERIALIZE` function (to work around a few nullability check things).
I'll make the `self` argument optional though.

Reply:
Hm, I suppose `self` isn't required for custom serializers, but it is required for the actual `__SERIALIZE` function (to work around a few nullability check things).
I'll make the `self` argument optional though.

Reply:
Created an issue to track it here: https://github.com/JairusSW/json-as/issues/115